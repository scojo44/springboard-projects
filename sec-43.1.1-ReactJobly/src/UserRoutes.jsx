import React, {useContext} from "react"
import {Navigate, Outlet} from "react-router-dom"
import UserContext from "./UserContext"

/** Wrapper for routes that require a login */

export default function UserRoutes() {
  const {currentUser} = useContext(UserContext);

  return currentUser? <Outlet /> : <Navigate to="/login" />;
}
