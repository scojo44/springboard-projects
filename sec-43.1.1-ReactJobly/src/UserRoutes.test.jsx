import {render} from '@testing-library/react'
import {MemoryRouter} from 'react-router-dom'
import UserContext from './UserContext'
import UserRoutes from './UserRoutes'

const context = {
  appliedJobIDs: new Set([1]),
  applyToJob: vi.fn(),
  currentUser: {
    username: 'testuser',
    firstName: 'First',
    lastname: 'Last',
    email: 'first@last.test',
    jobs: []
  }
};

describe('UserRoutes Tests', () => {
  it('Renders without crashing', () => {
    render(
      <MemoryRouter>
        <UserContext.Provider value={context}>
          <UserRoutes />
        </UserContext.Provider>
      </MemoryRouter>
    );
  });

  it('Matches snapshot', () => {
    const {asFragment} = render(
      <MemoryRouter>
        <UserContext.Provider value={context}>
          <UserRoutes />
        </UserContext.Provider>
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
