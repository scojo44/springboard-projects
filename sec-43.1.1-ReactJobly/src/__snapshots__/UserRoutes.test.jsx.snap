// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UserRoutes Tests > Matches snapshot 1`] = `<DocumentFragment />`;
